src/enemy.rs

```diff
@@ -0,0 +1,279 @@
+use std::collections::HashMap;
+use bevy::prelude::*;
+use crate::player::*;
+use std::time::Duration;
+use bevy::time::TimerMode::Repeating;
+use rand::{Rng, thread_rng};
+use crate::steering::{SteeringBundle, SteeringHost, SteerSeek};
+use crate::stats::*;
+use serde_json;
+use serde::{Deserialize, Serialize};
+
+pub struct EnemyPlugin;
+
+impl EnemyBundle {
+    fn new(h: Health, d: Damage, r: Rewards) -> Self {
+        Self {
+            enemy: Enemy,
+            health: h,
+            damage: d,
+            rewards: r,
+        }
+    }
+}
+/*
+todolist
+Walk towards player [V]
+Damage player when almost in the player (atm dying).
+Proper spawn of multiple enemies [V]
+Die. [V]
+ */
+
+// TODO: Move to config?
+const WAVE_DURATION_SEC:u64 = 30;
+const GLOBAL_TIME_TICKER_SEC:u64 = 5;
+impl Plugin for EnemyPlugin {
+    fn build(&self, app: &mut App) {
+        app
+            .insert_resource(CurrentWave{ num: 1, timer: Timer::new(Duration::from_secs(WAVE_DURATION_SEC ),Repeating), need_wave_spawn: true })
```

Comment:
it might be worth defining `Default` trait for `CurrentWave`

Reply:
Why?
Is it possible to re-declare default values for fields? 
All I can see from [doc](https://docs.rs/bevy/latest/bevy/prelude/fn.default.html), you can only use default values for types. 
In our case we need exact values at start

Maybe I'm not getting something?

Reply:
I think he means you should implement `Default` for `CurrentWave` so you can initialize it like this:
```rust
// snip
.insert_resource(CurrentWave::Default())
```

Reply:
yes and also use `..default()` if you need to override some field